### DISCUSSION TITLE
Run motor forever and acceleration question

### DISCUSSION THREAD

**Discussion by jeagh** [pybricks/support#906] [Q&A]
Two questions, if I may.

1. How do I run a motor forever rather than for a specified time? If I enclose the run method within a never-ending while loop then the motor will run forever but I don't think that's the correct way to have the motor run forever.

2. My understanding about how to reset the default acceleration or torque is a bit suspect as well. For example, if I reset the acceleration, as my example shows, then I don't notice any change in acceleration. Perhaps the acceleration is too subtle to notice? The motor seems to go from 0 to maximum without a gradual build up of speed.

```python
motor = Motor(Port.E)

motor.run(200)

data = motor.control.limits()

print('speed acceleration torque')
print(data)

# speed = 1080, acceleration = 2000 and torque = 199

wait(2000)
motor.stop()

wait(2000)

print('change acceleration')

motor.control.limits(acceleration=100) # torque=10 the torque doesn't seem to change either

motor.run(200)
wait(2000)

```

**Comment by laurensvalk** [comment]
Your program is correct, but a speed of just `200` deg/s is quite slow. (In terms of the MINDSTORMS block language, this is just "20%" speed).

So there isn't much to see in terms of acceleration. If you try

```python
motor.run(1000)
```

you should be able to see the difference in acceleration more clearly.

**Reply by jeagh** [reply]
Thank you laurensvalk, the acceleration is working correctly and Pybricks has renewed my interest in Lego projects.
